src/components/SignUp.jsx

```diff
@@ -0,0 +1,214 @@
+/* eslint-disable consistent-return */
+import React, { useState } from 'react'
+import { Box, Link, Typography, styled } from '@mui/material'
+import { useForm } from 'react-hook-form'
+import {
+   CloseModalIcon,
+   ContinueWithGoogle,
+   EyeClose,
+   EyeOpen,
+} from '../assets'
+import { Button } from './UI/Button'
+import { Input } from './UI/input/Input'
+import { Checkbox } from './UI/Checkbox'
+
+export const SignUp = () => {
+   const {
+      register,
+      handleSubmit,
+      formState: { errors },
+      watch,
+   } = useForm()
+
+   const onSubmit = () => {}
+
+   // password state
+
+   const [
+      visibleAndInvisiblePasswordState,
+      setVisibleAndInvisiblePasswordState,
+   ] = useState(false)
+
+   const changePasswordVisibleInvisibleStateHandler = () => {
+      setVisibleAndInvisiblePasswordState((prevState) => !prevState)
+   }
+
+   // confirm password state
+
+   const [
+      visibleAndInvisibleConfirmPasswordState,
+      setVisibleAndInvisibleConfirmPasswordState,
+   ] = useState(false)
+
+   const changeConfirmVisibleInvisibleStateHandler = () => {
+      setVisibleAndInvisibleConfirmPasswordState((prevState) => !prevState)
+   }
+   return (
+      <SignUpForm component="form">
+         <FormTitleAndCloseIcon>
+            <FormTitle variant="h4">Регистрация</FormTitle>
+            <CloseModalIcon style={{ cursor: 'pointer' }} />
+         </FormTitleAndCloseIcon>
+         <Input
```

Comment:
Render this three input through map. Because they have props and you can just create and array with object of input properties and then try to map it.